session1/examples/generics.ts

```diff
@@ -0,0 +1,33 @@
+//Generic type variables
+function checkPasswordLength<T>(password: T) {
+    console.log(password.length);
+}
+
+//Generic types
+function testArray<T>(item: Array<T>) {
+    return item;
+}
+let furits = testArray<string>(["Apples"]);
+furits.push(100);
+
+//Generic classes
+class stack<T>{
+    constructor(public item: Array<T>) {
+        }    
+}
+let vegetable = new stack<string>(["Lemons"]);
+vegetable.item.push(100);
+
+//Generic constraints
+class houeholdItems{
+    constructor(public nameOfItem: string) {       
+    }     
+}
+
+function cleaners<T extends houeholdItems>(c:T) {
```

Comment:
1. Similar comments!
2. The name of the function should be a verb otherwise it makes it a class or a variable.
3. Why are using ES5 function syntax here but not the ES6 arrow function?

Reply:
TBD